btclightclient/partial_merkle_tree.go

```diff
@@ -0,0 +1,215 @@
+package btclightclient
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"io"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// / We verify partial merkle tree proof. This logic are using in verifytxoutproof
+// / We are follow the logic in verifytxoutproof.
+
+type PartialMerkleTree struct {
+	numberTransactions uint
+	vBits              []bool
+	vHash              []*chainhash.Hash
+}
+
+type MerkleProof struct {
+	nodeValue  chainhash.Hash
+	merklePath []chainhash.Hash
+	pos        uint32
+}
+
+// TODO: verify this value
+const maxAllowBytes = 65536
+
+// TODO: wrap error
+func readPartialMerkleTree(r io.Reader, buf []byte) (*PartialMerkleTree, error) {
+	if _, err := io.ReadFull(r, buf[:4]); err != nil {
+		return nil, err
+	}
+
+	numberTransactions := uint(binary.LittleEndian.Uint32(buf[:4]))
+
+	var pver uint32 // pversion but btcd don't use this in those function we want.
+
+	var vHash []*chainhash.Hash
+	if numberHash, err := wire.ReadVarIntBuf(r, pver, buf); err != nil {
+		return nil, err
+	} else {
```

Comment:
`else` is not needed - we return in `if`

Reply:
Because we use value declared in if so, `else` is required. However, I fixed this ;)